site/source/tutorials/accessibility.md

```diff
@@ -0,0 +1,233 @@
+
+##### A11y properties
+- `headingLevel`: Optionally customize the heading level of the button controlling the accordion.
+___
+
+This overview only touched on properties that primarily exist for accessibility-related reasons, but any design decision or property change will end up affecting accessibility in some way. For some of those, such as `invalid`, `disabled`, and `readOnly`, we handle setting ARIA attributes in the background in addition to toggling classes without any extra attention needed from the user. In other cases, such as `getResultLabel` in ComboBox or `renderMonthLabel` in Calendar, it is entirely up to the widget user to ensure the returned result includes clear, screen-reader-accessible text content.
+
+## Styling
+The base Dojo 2 theme meets WCAG AA color contrast guidelines, but it is up to the user to double check the final styles as changes to font size or background could affect readability. Widget css packaged with Dojo2 also includes `:focus` styles for all focusable nodes.
+
+There is a set of base styles provided by `@dojo/widgets` separate from themes, containing basic utility classes like the screen reader-accessible visually hidden style. To use it, import `baseCss` separately like so:
+
+```js
+import * as css from ‘path/to/your/css’;
+import * as baseCss from ‘@dojo/widgets/common/styles/base.m.css’;
+
+@theme(css)
+class MyWidget extends WidgetBase {
+	render() {
+		return v(‘div’, {
+			classes: this.classes().fixed(baseCss.visuallyHidden)
+		}, [ ‘Screen reader instructions’ ]);
+	}
+}
+```
+
+## Focus Management
+Dojo 2 will provide a focus manager for situations where focus needs to be directly managed, since directly touching the DOM within widgets is discouraged. This feature is currently in development.
+
+## Writing Custom Widgets
+All DOM attributes can be set with `VirtualDomProperties` so, apart from managing focus, no extra tools should be required to create strongly accessible widgets. For example, the following code would create an accessible popup button:
+```js
+v('button', {
+	key: 'buttonKey',
+	'aria-controls': 'popupId',
+	'aria-expanded': String(this.properties.expanded),
+	'aria-haspopup': 'true',
+	id: 'buttonId',
+	onclick: this._togglePopup
+}, [ 'Click to open popup' ])
+```
+Since this example uses a `<button>` element, it requires neither an explicit `tabindex` nor an `onkeydown` event to work properly with a keyboard. Wherever possible, Dojo 2 takes advantage of built-in accessibility by using native elements; if it looks like a button and acts like a button, it should be a `<button>`. However, while native functionality is the easiest and most straightforward path to good accessibility, there are times when it’s just not possible or no such element exists.
```

Comment:
I wonder if the text starting from " Wherever possible, Dojo 2 takes advantage of built-in accessibility by using native elements;" to the end of the paragraph should be moved closer to the top of the article before "Using Dojo 2 Widgets".  It might be good if the reader knew some of these things before reading about each type of widget.